invenio_communities/models.py

```diff
@@ -311,11 +316,19 @@ def remove_record(self, record):
         :type record: `invenio_records.api.Record`
         """
         key = current_app.config['COMMUNITIES_RECORD_KEY']
+        access_key = '_access'
 
         assert self.id in record.get(key, [])
-
         record[key] = [c for c in record[key] if c != self.id]
 
+        assert self.id in record.get(access_key, {'read': []})['read']
+        record[access_key]['read'] = \
+            [c for c in record[access_key]['read'] if c != self.id]
```

Comment:
how complicated this is...
either you test if the value exist and then you remove it with the python given functions:
```python
read = record.get(access_key, {'read': []})
if self.id in read['read']
    read.remove(self.id)
```
the complexity is 2n as you need first to search for the value, and then to remove it... But the code is simple and beautiful, and in the end, you are in o(n).

OR you remove it with a comprehension list in order to avoid the search an have a complexity of O(n) for real:
```python
read = record.get(access_key, {'read': []})
read['read'] = [c for c in read['read'] if c != self.id]
```
not even sure this last solution is faster :/